lib/irslackd.js

```diff
@@ -432,7 +477,7 @@ class Irslackd {
     }
 
     // Set cache; return
-    ircNick = user.user.name;
+    ircNick = (user.user.profile.display_name) ? user.user.profile.display_name : user.user.name;
```

Comment:
Hm, I think `display_name` can contain spaces and wide chars. We could replace non-printable-ascii with an underscore or something, but then we'd risk duplicates. `display_name` might not be guaranteed unique across a workspace in the first place.